auth/detect/internal/stsexchange/sts_exchange.go

```diff
@@ -28,50 +28,70 @@ import (
 	"cloud.google.com/go/auth/internal"
 )
 
-type exchangeOptions struct {
-	client         *http.Client
-	endpoint       string
-	request        *stsTokenExchangeRequest
-	authentication clientAuthentication
-	headers        http.Header
-	extraOpts      map[string]interface{}
+const (
+	// GrantType for a sts exchange.
+	GrantType = "urn:ietf:params:oauth:grant-type:token-exchange"
+	// TokenType for a sts exchange.
+	TokenType = "urn:ietf:params:oauth:token-type:access_token"
+
+	jwtTokenType = "urn:ietf:params:oauth:token-type:jwt"
+)
+
+// Options stores the configuration for making an sts exchange request.
+type Options struct {
+	Client         *http.Client
+	Endpoint       string
+	Request        *TokenRequest
+	Authentication ClientAuthentication
+	Headers        http.Header
+	ExtraOpts      map[string]interface{}
```

Comment:
Can you add a comment explaining `ExtraOpts` and what goes in it? (Or if it's already explained somewhere else, copy the explanation to here?)

Reply:
done.